deepwell/src/services/page/service.rs

```diff
@@ -565,6 +565,28 @@ impl PageService {
         Ok(page)
     }
 
+    /// Gets all deleted pages that match the provided slug.
+    pub async fn get_deleted_slug(
+        ctx: &ServiceContext<'_>,
+        site_id: i64,
+        slug: String,
+    ) -> Result<Vec<PageModel>> {
+        let txn = ctx.transaction();
+        let pages = {
+            Page::find()
+                .filter(
+                    Condition::all()
+                        .add(page::Column::Slug.eq(trim_default(&slug)))
+                        .add(page::Column::SiteId.eq(site_id))
+                        .add(page::Column::DeletedAt.is_not_null()),
+                )
```

Comment:
Don't assume database insertion order will be good, I would add an explicit `ORDER BY created_at DESC`.